### PR TITLE
Feature - TuanTran - Swift OOP

### PR DESCRIPTION
**Pull request by mvn-tuantran-dn**
- [x] Bai 1
- [x] Bai 2
- [x] Bai 3
- [x] Bai 4
- [x] Bai 5
- [x] Bai 6
- [x] Bai 7
- [x] Bai 8
- [x] Bai 9
- [x] Bai 10

### CODE REVIEW

File: 03_SwiftOOP/TuanTV/BT-OOP-BT2.playground/Contents.swift [hunk 1]
```diff
@@ -0,0 +1,221 @@
+import UIKit
+
+class DaGiac {
+    var soCanhDaGiac: Int
+    var mangDoLonCanhDaGiac: [Int]
+    init(n: Int, mangDoLon: [Int]) {
+        self.soCanhDaGiac = n
+        self.mangDoLonCanhDaGiac = Array(mangDoLon[0..<n])
+    }
+    func tinhChuVi() -> Int{
```
Comment (mvn-toannguyen4-dn) [inline 03_SwiftOOP/TuanTV/BT-OOP-BT2.playground/Contents.swift:10]: nên Int {

File: 03_SwiftOOP/TuanTV/BT-OOP-BT2.playground/Contents.swift [hunk 2]
```diff
@@ -0,0 +1,221 @@
+import UIKit
+
+class DaGiac {
+    var soCanhDaGiac: Int
+    var mangDoLonCanhDaGiac: [Int]
+    init(n: Int, mangDoLon: [Int]) {
+        self.soCanhDaGiac = n
+        self.mangDoLonCanhDaGiac = Array(mangDoLon[0..<n])
+    }
```
Comment (mvn-toannguyen4-dn) [inline 03_SwiftOOP/TuanTV/BT-OOP-BT2.playground/Contents.swift:9]: nên cách 1 dòng sau mỗi function

File: 03_SwiftOOP/TuanTV/BT-OOP-BT2.playground/Contents.swift [hunk 3]
```diff
@@ -0,0 +1,221 @@
...
+        
+    }
+    func inGiaTriCacCanh(){
+        for (index,doLonCanh) in mangDoLonCanhDaGiac.enumerated() {
+            print("do lon cua canh thu \(index) = \(doLonCanh)")
+        }
+    }
+}
+
+DaGiac.init(n: 3, mangDoLon: [1,2,3]).inGiaTriCacCanh()
```
Comment (mvn-toannguyen4-dn) [inline 03_SwiftOOP/TuanTV/BT-OOP-BT2.playground/Contents.swift:25]: [1,2,3] -> [1, 2, 3]

File: 03_SwiftOOP/TuanTV/BT-OOP-BT2.playground/Contents.swift [hunk 4]
```diff
@@ -0,0 +1,221 @@
...
+    var top: Int
+    var stack: [Int]
+    init(size: Int, top: Int, stack: [Int]) {
+        self.size = size
+        self.top = top
+        self.stack = stack
+    }
+    
+    func isEmpty() -> Bool {
+        return top == 0 ? true : false
```
Comment (mvn-toannguyen4-dn) [inline 03_SwiftOOP/TuanTV/BT-OOP-BT2.playground/Contents.swift:65]: code dư

File: 03_SwiftOOP/TuanTV/BT-OOP-BT2.playground/Contents.swift [hunk 5]
```diff
@@ -0,0 +1,221 @@
...
+class TamGiac: DaGiac {
+    
+    func tinhDienTichTamGiac() {
+        var chuVi: Float = Float(tinhChuVi())
+        chuVi = sqrt(chuVi * (chuVi - Float(mangDoLonCanhDaGiac[0])) * (chuVi - Float(mangDoLonCanhDaGiac[1])) * (chuVi - Float(mangDoLonCanhDaGiac[2])))
+    }
+    
+    func inCacCanhTamGiacPitago(dsTamGiac: [TamGiac]) {
+        for tamGiac in dsTamGiac {
+            var canhA = 0
```
Comment (mvn-toannguyen4-dn) [inline 03_SwiftOOP/TuanTV/BT-OOP-BT2.playground/Contents.swift:36]: var canhA = 0
canhA = tamGiac.mangDoLonCanhDaGiac[0]
--> let canhA = tamGiac.mangDoLonCanhDaGiac[0]

File: 03_SwiftOOP/TuanTV/BT-OOP.playground/Contents.swift [hunk 6]
```diff
@@ -0,0 +1,118 @@
...
+        
+        guard  let thuong = chiaHaiPhanSo(phanSo1, phanSo2)
+        else {
+            print("phan so co mau = 0 la phan so khong xac dinh")
+            return
+        }
+        print("tich 2 phan so = " + rutGonPhanSo(phanSo: thuong).inPhanSo())
+    }
+}
+var phanso = PhanSo.init(tuso: 1, mauso: 1)
```
Comment (mvn-toannguyen4-dn) [inline 03_SwiftOOP/TuanTV/BT-OOP.playground/Contents.swift:113]: phanso -> phanSo

File: 03_SwiftOOP/TuanTV/BT-OOP.playground/Contents.swift [hunk 7]
```diff
@@ -0,0 +1,118 @@
+import UIKit
+
+//MARK: -BAI TAP 1
+class PhanSo {
+    //MARK: prototype
+    var tuso: Int
```
Comment (mvn-toannguyen4-dn) [inline 03_SwiftOOP/TuanTV/BT-OOP.playground/Contents.swift:6]: tuso -> tuSo

File: 03_SwiftOOP/TuanTV/BT-OOP-BT2.playground/Contents.swift [hunk 8]
```diff
@@ -0,0 +1,299 @@
...
+        return 2 * pi * banKinh
+    }
+    override func tinhDienTich() -> Float {
+        return 4 * pi * banKinh * banKinh
+    }
+    override func tinhTheTich() -> Float {
+        return 4 / 3 * pi * banKinh * banKinh * banKinh
+    }
+}
+class HinhlapPhuong: BaChieu {
```
Comment (mvn-toannguyen4-dn) [inline 03_SwiftOOP/TuanTV/BT-OOP-BT2.playground/Contents.swift:175]: HinhlapPhuong -> HinhLapPhuong

**Review by mvn-thinhnguyen2-dn — APPROVED** [review]
LGTM!

**Pull request merged**